searx/engines/radio_browser.py

```diff
@@ -52,18 +54,33 @@
 
 """
 
-servers = []
+CACHE: EngineCache
+"""Persistent (SQLite) key/value cache that deletes its values after ``expire``
+seconds."""
 
 
 def init(_):
-    # see https://api.radio-browser.info
+    global CACHE  # pylint: disable=global-statement
+    CACHE = EngineCache("radio_browser")
+    server_list()
+
+
+def server_list() -> list[str]:
+
+    servers = CACHE.get("servers", [])
+    if servers:
+        return servers
+
     ips = socket.getaddrinfo("all.api.radio-browser.info", 80, 0, 0, socket.IPPROTO_TCP)
     for ip_tuple in ips:
         _ip: str = ip_tuple[4][0]  # type: ignore
         url = socket.gethostbyaddr(_ip)[0]
         srv = "https://" + url
         if srv not in servers:
             servers.append(srv)
+    # update server list once in 24h
+    CACHE.set(key="servers", value=servers, expire=60 * 60 * 24)
+    return servers
```

Comment:
Maybe we should add a check here to throw/show a proper error message if `servers` is empty? Otherwise `random.choice` will just crash where this method is called.